test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+  - script: |
+      set -e
+      set -o pipefail
+
+      # These variables need to be maintained according to the component matrix: https://spaces.redhat.com/display/GITOPS/GitOps+Component+Matrix
+      expected_kustomizeVersion='v5.0.1'
+      expected_helmVersion='v3.11.2'
+      expected_argocdVersion='v2.7.6'
+      expected_dexVersion='v2.35.1'
+      expected_redisVersion='6'
+
+      MAX_RETRIES=10
+      get_pod_name() {
+        local retry_count=0
+        until [ "${retry_count}" -gt ${MAX_RETRIES} ]; do
+          retry_count=$((retry_count + 1))
+          pod=$(
+            oc get -n openshift-gitops pods \
+              -o custom-columns=NAME:.metadata.name --no-headers |
+              grep $1 |
+              head -1
+          )
+          if [[ -z "${pod}" ]]; then
+            #// no set
+            continue
+          else
+            #//set
+            echo $pod
+            break
+          fi
+        done
+      }
+
+      gitops_server_pod=$(get_pod_name openshift-gitops-server)
+
+      dex_pod=$(get_pod_name openshift-gitops-dex-server)
+       
+      redis_pod=$(get_pod_name openshift-gitops-redis)
+
+      route=$(
+        oc get route -n openshift-gitops \
+          -o custom-columns=NAME:.spec.host --no-headers |
+          grep openshift-gitops-server |
+          head -1
+      )
+
+      kustomizeVersion=$(
+        oc -n openshift-gitops exec $gitops_server_pod \
+          -- kustomize version
+      )
+      helmVersion=$(
+        oc -n openshift-gitops exec $gitops_server_pod \
+          -- helm version |
+          sed -e 's/version.BuildInfo//' -e 's/\"//g' |
+          awk -F':' '{ print $2 }' |
+          awk -F',' '{ print $1 }'
+      )
+      argocdVersion=$(
+        oc -n openshift-gitops exec $gitops_server_pod -- \
+          argocd version --short --server $route --insecure |
+          grep 'argocd-server' |
+          sed -e 's/\+unknown//' |
+          awk -F' ' '{ print $2 }'
+      )
+      dexVersion=$(
+        oc -n openshift-gitops exec $dex_pod -- dex version 2>&1 |
+          grep '^Dex Version' |
+          awk -F': ' '{ print $2 }'
+      )
+
+      redisVersion=$(
+        oc -n openshift-gitops exec $redis_pod -- \
+          redis-server -v | awk -F '=' '{ print $2 }' | cut -d'.' -f 1
+      )
+
+      if test "${kustomizeVersion}" != "${expected_kustomizeVersion}"; then
+        echo "Kustomize version mismatch. Should be ${expected_kustomizeVersion}, is ${kustomizeVersion}"
+        exit 1
```

Comment:
Hi @anandrkskd 
Lets  catch **exit 1** after setting a  flag for each condition if a mismatch happens